### ISSUE TITLE
Dafny does not have a binary release for Linux arm64

### ISSUE THREAD

**Issue by tchajed** [dafny-lang/dafny#3972]
### Summary

I'd like to propose adding Linux arm64 as another platform with a binary release.

### Background and Motivation

For a course using Dafny, we wanted to tell everyone to run Dafny in a Docker devcontainer. On an M1 mac, this ends up needing Dafny to run in a Linux arm64 VM, and there's no binary release for that platform.

If you're a researcher, this isn't an issue: you just install Dafny natively using Homebrew and don't run it inside the container. However, for a class it's nice to have a single, consistent set of instructions that work for everyone.

It's also possible to end up with an emulated x86 VM as the devcontainer and then the usual binary release works, but I don't fully understand this (it didn't work for all of our students, and it did seem noticeably slow).

### Proposed Feature

Add Ubuntu arm64 as a platform to the relevant release script.

### Alternatives

This is mostly to have a really streamlined process for students. If you're committed to using Dafny (eg, you're a researcher) you can install and use it through other means.

**Comment by tchajed** [comment]
I should mention that I don't have an M1 mac so my understanding of this issue is limited to what I've seen on other people's computers, which was quite frankly confusing.

**Comment by tchajed** [comment]
I've now had a chance to test out Dafny myself on an M1 mac. Using Dafny through Homebrew works, except that I had to manually provide the path to dotnet@6 (`/opt/homebrew/Cellar/dotnet@6/6.0.114/bin/dotnet`).

When using an Ubuntu arm64 Docker container, VS Code is unusable because the extension insists on downloading Dafny and this fails because <https://github.com/dafny-lang/dafny/releases/download/v4.2.0/dafny-4.2.0-arm64-ubuntu-20.04.zip> isn't released. This is a huge shame because a Docker container would have given students an isolated and easily reproduced dev environment.

**Comment by Xazax-hun** [comment]
It would be tremendously useful to have arm64 binaries for Linux (and maybe for windows as well).

**Comment by Robert42** [comment]
> [...] I see that Z3 tag was added. I believe Z3 now has arm64 binaries released for Linux.

Yes, Z3 has arm64 binaries released for linux, but those are currently broken:
- https://github.com/Z3Prover/z3/issues/7201
- Apparently not the first time: https://github.com/Z3Prover/z3/issues/6835

**Comment by Robert42** [comment]
Z3 has now arm64 builds
